### ISSUE TITLE
User Role Manager corrupted D365 environment!

### ISSUE THREAD

**Issue by khendricks89** [MscrmTools/MsCrmTools.SecurityRelated#9]
We used the "User Role Manager" on a D365 v9.0.2.1470 and it corrupted our environment! The updater should have removed a custom Security Role that is derived from the "Field Service - Inventory Purchase" role, but instead all users independent of the rights, even the System Administrator role lost the "prv_readsav_viewroleconfiguration" right. This is a special core right that can't be set via the UI. The system is now corrupted and needs to be reviewed by Microsoft.

Please disable the plugin untill a fix has been found to avoid others from have a simular issue.

**Comment by khendricks89** [comment]
![image](https://user-images.githubusercontent.com/5470587/44979860-89534800-af6e-11e8-9ecd-4975a7a157f4.png)

**Comment by khendricks89** [comment]
We already identified that this is not the only right that has been removed. We are unable to use most of the D365 administration functionality.

**Comment by MscrmTools** [comment]
Can you precise what action you did exactly with this plugin that led to this problem?

**Comment by khendricks89** [comment]
Of course. We used the newest version of the XRMToolbox and User Role Manager plugin to remove a security role from all users. Please see below for more details of the exact selection. In the plugin we used the second option under Actions.

![image](https://user-images.githubusercontent.com/5470587/44985071-4352b000-af7f-11e8-9588-b915dd9acf33.png)

![image](https://user-images.githubusercontent.com/5470587/44985038-2f0eb300-af7f-11e8-8629-23898a976b64.png)

**Comment by khendricks89** [comment]
I asure you. I ONLY removed the roles as described with this plugin and the minute after every user (including the administrator on the top that is not selected) have lost core rights. I understand that this plugin does not change individual rights, that being said. The system worked fine for the last 6 months, and the minute this operation is done the system is corrupt. No other users where in this development system accept myself, no other operations are being done in the background.

Fair or not, the plugin triggered the issue. Does not mean it is not an issue within the D365 framework / instance that causes the end result, but the plugin initiated it.

**Comment by MscrmTools** [comment]
Do you have a support ticket number for that issue? I can contact the product team directly to ask for help so that they can tell me if there is a bug they are aware of regarding this (strange) behavior

**Comment by khendricks89** [comment]
That would be #118090318915288 . I reached out to some support engineers I worked with over the last week to esclate this issue into the highest priority as our deadline for the freeze is wednesday and release is in 3 weeks. Currently the issue has not been assigned.

**Comment by MscrmTools** [comment]
ok, let me try to make it progress

**Comment by khendricks89** [comment]
Thank you, that is very much appreciated.

**Comment by MscrmTools** [comment]
Just sent the email to the product team. I don't have any ETA when I will have an answer. I keep you posted

**Comment by khendricks89** [comment]
We are assigned a support engineer from Microsoft for the case. See did not recieve the data from the product team yet, in case you can/wish to send her more details, her contact is: Sonam Patel v-9sopat@microsoft.com referencing [REG:118090318915288] .

**Comment by MscrmTools** [comment]
I got no answer so far, I will try to add this information. 
Thanks
